static/app/views/replays/detail/console/index.tsx

```diff
@@ -15,21 +16,45 @@ interface Props {
   breadcrumbs: BreadcrumbTypeDefault[];
 }
 
+const getDistinctLogLevels = breadcrumbs =>
+  Array.from(new Set(breadcrumbs.map(breadcrumb => breadcrumb.level)));
+
 function Console({breadcrumbs}: Props) {
   const [searchTerm, setSearchTerm] = useState('');
+  const [logLevel, setLogLevel] = useState<BreadcrumbLevelType[]>([]);
   const handleSearch = debounce(query => setSearchTerm(query), 150);
+
   const filteredBreadcrumbs = useMemo(
     () =>
-      !searchTerm
+      !searchTerm && logLevel.length === 0
         ? breadcrumbs
-        : breadcrumbs.filter(breadcrumb =>
-            breadcrumb.message?.toLowerCase().includes(searchTerm)
+        : breadcrumbs.filter(
+            breadcrumb =>
+              breadcrumb.message?.toLowerCase().includes(searchTerm) &&
+              logLevel.includes(breadcrumb.level)
           ),
-    [searchTerm, breadcrumbs]
+    [logLevel, searchTerm, breadcrumbs]
   );
+
   return (
     <Fragment>
-      <ConsoleSearch onChange={handleSearch} />
+      <ConsoleFilters>
+        <CompactSelect
+          triggerProps={{
+            prefix: t('Log Level'),
+          }}
+          multiple
+          options={getDistinctLogLevels(breadcrumbs).map(breadcrumbLogLevel => ({
+            value: breadcrumbLogLevel,
+            label: breadcrumbLogLevel as string,
```

Comment:
i don't think `as string` is the right fix

Either convert to string explicitly `label: String(breadcrumbLogLevel),`

Remember that `String(undefined) === "undefined"`. So the fix might really be to make `breadcrumbLogLevel` be typed as `string` and not `string | undefined` at its source.

Reply:
I put a type of string on the set of log levels, BreadcrumbLevelType doesn't have undefined as a possible value so the issue was just local to my implementation.